server/src/main/java/org/opensearch/extensions/ExtensionsSettings.java

```diff
@@ -43,6 +43,7 @@ public static class Extension {
         private String className;
         private String customFolderName;
         private String hasNativeController;
+        private List<ExtensionDependency> dependencies;
```

Comment:
Rereading this again, I see this is static inner `Extension` class here in `ExtensionSettings`.  I don't think this is ever used or is needed at all.  It's completely redundant with the `DiscoveryExtension` class which is used elsewhere.  

The change to add the list of dependencies is done in the `DiscoveryExtension` class.

I actually think this entire inner class can be deleted.  @owaiskazi19  do you agree?

Reply:
@dbwiddis Can we use `List.of(new ExtensionDependency(extension.getName(), Version.fromString(extension.getOpensearchVersion())))` to replace `extension.getDepdendency` in `ExtensionsOrchestrator`, in this way, we can delete the `getDependency` method in `ExtensionSetting`.

Reply:
> @dbwiddis Can we use `Arrays.asList(...`

It's been a while since I did YAML import from Jackson.  If Jackson reads in a list as an array, then:
 - define the dependencies in this class as an array
 - in ExtensionsOrchestrator, convert the array to a list as `Arrays.asList(extension.getDependencies())`